lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -905,40 +905,41 @@ namespace Js
 
         JavascriptString* accumulatedResult = nullptr;
 
-        BEGIN_TEMP_ALLOCATOR(tempAlloc, scriptContext, _u("RegexHelper"))
-        {
-            JsUtil::List<RecyclableObject*, ArenaAllocator>* results =
-                JsUtil::List<RecyclableObject*, ArenaAllocator>::New(tempAlloc);
+        Recycler* recycler = scriptContext->GetRecycler();
 
-            while (true)
-            {
-                PCWSTR varName = _u("RegExp.prototype[Symbol.replace]");
-                Var result = JavascriptRegExp::CallExec(thisObj, input, varName, scriptContext);
-                if (JavascriptOperators::IsNull(result))
-                {
-                    break;
-                }
+        JsUtil::List<RecyclableObject*>* results = RecyclerNew(recycler, JsUtil::List<RecyclableObject*>, recycler);
 
-                RecyclableObject* resultObj = ExecResultToRecyclableObject(result);
+        while (true)
+        {
+            PCWSTR varName = _u("RegExp.prototype[Symbol.replace]");
+            Var result = JavascriptRegExp::CallExec(thisObj, input, varName, scriptContext);
+            if (JavascriptOperators::IsNull(result))
+            {
+                break;
+            }
 
-                results->Add(resultObj);
+            RecyclableObject* resultObj = ExecResultToRecyclableObject(result);
 
-                if (!global)
-                {
-                    break;
-                }
+            results->Add(resultObj);
 
-                JavascriptString* matchStr = GetMatchStrFromResult(resultObj, scriptContext);
-                AdvanceLastIndex(thisObj, input, matchStr, unicode, scriptContext);
+            if (!global)
+            {
+                break;
             }
 
-            CompoundString::Builder<64 * sizeof(void *) / sizeof(char16)> accumulatedResultBuilder(scriptContext);
-            CharCount inputLength = input->GetLength();
-            CharCount nextSourcePosition = 0;
+            JavascriptString* matchStr = GetMatchStrFromResult(resultObj, scriptContext);
+            AdvanceLastIndex(thisObj, input, matchStr, unicode, scriptContext);
+        }
 
-            size_t previousNumberOfCapturesToKeep = 0;
-            Var* captures = nullptr;
+        CompoundString::Builder<64 * sizeof(void *) / sizeof(char16)> accumulatedResultBuilder(scriptContext);
+        CharCount inputLength = input->GetLength();
+        CharCount nextSourcePosition = 0;
+
+        size_t previousNumberOfCapturesToKeep = 0;
+        Var* captures = nullptr;
 
+        BEGIN_TEMP_ALLOCATOR(tempAlloc, scriptContext, _u("RegexHelper"))
```

Comment:
It's used to create an array of String indexes for the `ReplaceFormatString` function and won't be GC'd like `results`.